### PR TITLE
Add video field to listing page

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonicalltd/snap-squad/issues/785

## Done
- Added a Video field
- Ensured API errors are displayed above the field
- Added temporary youtube output to snap details page for testing

## QA
- Pull the branch
- `./run`

### Setting a valid url
- Visit http://0.0.0.0:8004/:snap_name:/listing
- Add a video URL in the correct format
- Save
- Should succeed

### Setting an invalid url
- Visit http://0.0.0.0:8004/:snap_name:/listing
- Add a random string to the video field
- Save
- Should error with a message about formatting

### Testing propagation to public pages
- Visit http://0.0.0.0:8004/:snap_name:/listing
- Add a valid youtube url
- Save
- Visit http://0.0.0.0:8004/:snap_name:
- The first item in the carousel should be the video

### CODE REVIEW

**Comment by evandandrea** [comment]
Works here. Added a YouTube video, then a screenshot, then deleted both. All worked as expected.

**Comment by bartaz** [comment]
We mention asciicinema, but it's not supported yet (?) - or at least doesn't seem to work.
Also, URL should be all caps:

screen shot 2018-12-14 at 09 31 46

**Comment by bartaz** [comment]
> The public page is there for testing only. My next task is to make the front work properly.

OK. But if we want to merge this PR to master it should be 'production-ready'. So it's fine if it's just for testing, but it should be removed or fixed before this this is merged.

**Comment by Lukewh** [comment]
Commented out the video part @bartaz

File: templates/store/snap-details.html [hunk 1]
```diff
@@ -103,18 +103,18 @@ <h1 class="p-heading--two p-snap-heading__name">{{ snap_title }}</h1>
...
         <div class="p-carousel" id="js-snap-screenshots">
           <div class="swiper-container">
             <div class="swiper-wrapper">
-              {% for video in videos %}
-                <div class="p-carousel__item--video swiper-slide">
-                  <div class="js-video-slide" data-video-type="{{video.type}}" data-video-url="{{video.url}}">
-                  </div>
-                </div>
-              {% endfor %}
+              <!--{% for video in videos %}-->
```
Comment (bartaz) [inline templates/store/snap-details.html]: Can we comment it out from template `{# .... #}` rather then still rendering it into HTML?

File: templates/publisher/listing.html [hunk 2]
```diff
@@ -214,6 +214,36 @@
           </div>
         </div>
 
+        <div class="row">
+          <div class="col-8">
+            <div class="p-form-validation {% if field_errors and field_errors['video_urls'] %}is-error{% endif %}">
+              <label>Video: </label>
```
[the remark this answers is not in the document]
Comment (Lukewh) [reply templates/publisher/listing.html]: Interesting. All the discussions around if have described it as "Video". IMO Media = video, images and audio so "Media" would conflict a little with "Screenshots" above.

asciinema presents as a video with a play button, so I'd say it's video.

[the remark this answers is not in the document]
Comment (bartaz) [reply templates/publisher/listing.html]: OK.

**Pull request merged**